src/components/PostDetails/PostDetails.jsx

```diff
@@ -1,45 +1,80 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
 import { NewCommentForm } from '../NewCommentForm';
 import './PostDetails.scss';
+import { getPostDetails } from '../../api/posts';
+import { getPostComments, deletePostComment } from '../../api/comments';
 
-export const PostDetails = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
+export const PostDetails = ({ selectedPostId }) => {
+  const [selectedDetails, setSelectedDetails] = useState(null);
+  const [comments, setComments] = useState(null);
+  const [isHidden, setIsHidden] = useState(false);
+
+  useEffect(() => {
+    getPostDetails(selectedPostId)
+      .then(details => setSelectedDetails(details));
+
+    getPostComments(selectedPostId)
+      .then(result => setComments(result));
+  }, [selectedPostId, comments]);
+
+  const hide = () => {
+    setIsHidden(!isHidden);
+  };
+
+  return (
+    selectedDetails && (
+      <div className="PostDetails">
+        <h2>Post details:</h2>
+
+        <section className="PostDetails__post">
+          <h3>{selectedDetails.title}</h3>
+          <p>{selectedDetails.body}</p>
+        </section>
+
+        <section className="PostDetails__comments">
+          {!comments.length || (
```

Comment:
```suggestion
          {comments.length && (
```
в conditinal rendering використовуй &&

Reply:
done